SCloudRandomizer/MusicSource.m

```diff
@@ -45,15 +45,21 @@ - (void)getRandomTrack:(SearchParams *)searchParams completionHandler:(singleTra
     NetworkStatus networkStatus = [networkReachability currentReachabilityStatus];
     if (networkStatus == NotReachable) {
         NSLog(@"There is no internet connection");
-        completionHandler(nil, NoData);
+        completionHandler(nil, NoConnection);
     } else {
         [self getTracks:searchParams completionHandler:^(NSArray* tracks, enum MusicSourceError error) {
             if (error != None) {
                 completionHandler(nil, error);
             } else {
                 int randomSongIndex = arc4random_uniform((uint32_t) tracks.count);
-                Track* track = [[Track alloc] initWithData:[tracks objectAtIndex:randomSongIndex] account:[MusicSource account]];
-                completionHandler(track, None);
+                @try {
+                    Track* track = [[Track alloc] initWithData:[tracks objectAtIndex:randomSongIndex] account:[MusicSource account]];
+                    completionHandler(track, None);
+                }
+                @catch (NSException * e) {
+                    NSLog(@"Exception: %@", e);
+                    completionHandler(nil, TrackError);
+                }
```

Comment:
Yeah I once hit an exception with Track data in the initializer. Just a safety net if it happens again.